src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -496,6 +497,78 @@ public boolean impliesClusterPermissionPermission(String action) {
             return roles.stream().filter(r -> r.impliesClusterPermission(action)).count() > 0;
         }
 
+        /**
+         * Checks if the route is accessible via legacy naming convention.
+         * We check against cluster permissions because the legacy convention for cluster permissions map 1-1 with a transport
+         * action call initiated via REST API handler. Hence we use the same to allow/block request forwarding to ext.
+         * This ensures backwards-compatibility
+         *
+         * This assumes the convention that every new route register will be of type `plugin:routeName` and will then c
+         *
+         * NOTE: THIS CHECK WILL BE REMOVED ONCE ALL ACTIONS HAVE BEEN MIGRATED TO THE NEW CONVENTION
+         *
+         * E.g For ext `hw`, following are two possible ways actions an be defined in roles:
+         *
+         * ext_hw_greet:
+         *   reserved: true
+         *   cluster_permissions:
+         *     - 'hw:greet'
+         *
+         * legacy_hw_greet:
+         *   reserved: true
+         *   cluster_permissions:
+         *     - 'cluster:admin/opensearch/hw/greet'
+         *
+         *
+         * @param action - The action to be checked against its legacy version
+         * @return true, if a legacy version was found and validated, false otherwise
+         */
+        @Override
+        public boolean impliesLegacyPermission(String action) {
```

Comment:
Is this still needed now that action names can be associated directly in a NamedRoute? IMO I'd like to see this removed.

Reply:
I will remove this code, but that will mean that action names will have to be passed in order for legacy check to work.

Reply:
IMO it makes sense to put that burden on the extension/plugin dev and not try to formulate what the legacy transport action name is associated with what is being converted.

When Anomaly Detection plugin is being converted to an extension, the extension dev would have to convert each route handler to using a NamedRoute to afford it the same protections they have in place today with transport action names. If they want to keep their existing roles compatible with how they are defined as a plugin today then they would register the `actionNames` list of `NamedRoute` and the evaluator introduced in this PR would evaluate permissions against the `actionNames` registered with the route. There is no inferring the name of a transport action name.